roc/log.go

```diff
@@ -183,6 +189,40 @@ func loggerRoutine() {
 	}
 }
 
+func logWrite(level LogLevel, text string, params ...interface{}) {
+	storedLogLevel := LogLevel(atomic.LoadInt32(&loggerLevel))
+	if level > storedLogLevel {
+		return
+	}
+
+	var finalPath string
+	_, file, line, _ := runtime.Caller(1)
+
+	dirs := strings.FieldsFunc(file, func(c rune) bool {
+		return c == '/' || c == '\\'
+	})
+	fileNameIndex := len(dirs) - 1
+	lastDirIndex := len(dirs) - 2
+
+	if lastDirIndex >= 0 {
+		fileName := dirs[fileNameIndex]
+		lastDir := dirs[lastDirIndex]
+
+		finalPath = filepath.Join(lastDir, fileName)
+	}
```

Comment:
This is the same algorithm that you already implemented in previous versions, just expressed in different way. You're still implicitly hard-coding specific nesting level (2). Please read carefully what algorithm I suggested as an alternative and why, or feel free to suggest other algorithms that avoid hard-coding nesting level. https://github.com/roc-streaming/roc-go/pull/80#issuecomment-1557835660